src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/controllers/web/WorkflowRunControllerTests.kt

```diff
@@ -311,6 +311,63 @@ class WorkflowRunControllerTests
         )
     }
 
+    @Test
+    fun `can get the status of a workflow`()
+    {
+        val context = mock<ActionContext> {
+            on { params(":key") } doReturn "workflow_key1"
+            on { userProfile } doReturn CommonProfile().apply { id = "test@user.com" }
+        }
+
+        val mockAPIResponseText = """{"data": {"workflow_key": "workflow_key1", "status": "success"}}"""
+
+        val mockAPIResponse = OrderlyServerResponse(mockAPIResponseText, 200)
+
+        val apiClient = mock<OrderlyServerAPI> {
+            on { get(any(), any<Map<String, String>>()) } doReturn mockAPIResponse
+        }
+
+        val repo = mock<WorkflowRunRepository>()
+        val sut = WorkflowRunController(context, repo, apiClient)
+        val result = sut.getWorkflowRunStatus()
+
+        verify(apiClient).get("/v1/workflow/workflow_key1/status/", emptyMap())
+
+        assertThat(
+            Serializer.instance.gson.fromJson(
+                JsonLoader.fromString(result)["data"].toString(),
+                WorkflowRunController.WorkflowRunStatusResponse::class.java
+            )
+        ).isEqualTo(
+            WorkflowRunController.WorkflowRunStatusResponse("workflow_key1", "success")
+        )
+
+        verify(repo).updateWorkflowRun("workflow_key1", "success")
+    }
+
+    @Test
+    fun `passes through error when getting status of a workflow`()
+    {
+        val context = mock<ActionContext> {
+            on { params(":key") } doReturn "workflow_key1"
+            on { userProfile } doReturn CommonProfile().apply { id = "test@user.com" }
+        }
+
+        val mockResponse = """{"status": "failure", "data": null, "errors": []}"""
+
+        val apiClient = mock<OrderlyServerAPI> {
+            on { get(any(), any<Map<String, String>>()) } doReturn OrderlyServerResponse(
+                mockResponse,
+                400
+            )
+        }
+
+        val sut = WorkflowRunController(context, mock(), apiClient)
+        val response = sut.getWorkflowRunStatus()
+        verify(context).setStatusCode(400)
```

Comment:
I suppose for completeness you could add a test that the repo doesn't get called to update the workflow - though actually I guess you know that it hasn't because it would have failed to deserialize the reponse and thrown an error..